components/supervisor-api/generate-java.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
```

Comment:
It would be nice that we don't need another file. Calling `generate.sh` should do everything.

Reply:
It should, but as it also compiles Go and Typescript files, it takes much longer (15-30 seconds) (and there's also a non-null chance of failing while downloading Go packages).

I could make them only a single file later if the engineering team decides to remove also Go and TypeScript files from versioning.

Reply:
generate.sh could call generate-java.sh that it works as before

Reply:
Yes, generate.sh is already calling generate-java.sh, so its behaviour remains the same:

<img width="1418" alt="image" src="https://user-images.githubusercontent.com/418083/209123669-e046752e-d586-4d08-a6b9-5e62a78ac864.png">

The reason I left the java compilation in a separate file is for us to be able to call it individually (saving several seconds in the build):

<img width="541" alt="image" src="https://user-images.githubusercontent.com/418083/209123815-bb7168b3-b7d7-40de-8b1e-ec871f7e1fd3.png">

<img width="460" alt="image" src="https://user-images.githubusercontent.com/418083/209123866-d0994911-3019-4a31-91b4-7490127cb133.png">

Reply:
sounds good